src/engine/enginemaster.cpp

```diff
@@ -632,41 +634,42 @@ void EngineMaster::process(const int iBufferSize) {
-            if (m_pNumMicsConfigured->get() > 0
-                && !m_bExternalRecordBroadcastInputConnected) {
-                // Copy the master mix to a separate buffer before delaying it
-                // to avoid delaying the master output.
-                m_pLatencyCompensationDelay->process(m_pSidechainMix, m_iBufferSize);
-                SampleUtil::add(m_pSidechainMix, m_pTalkover, m_iBufferSize);
+                if (m_pNumMicsConfigured->get() > 0) {
+                    // The talkover signal Mixxx receives is delayed by the round trip latency.
+                    // There is an output latency between the time Mixxx processes the audio
+                    // and the user hears it. So if the microphone user plays on beat with
+                    // what they hear, they will be playing out of sync with the engine's
+                    // processing by the output latency. Additionally, Mixxx gets input signals
+                    // delayed by the input latency. By the time Mixxx receives the input signal,
+                    // a full round trip through the signal chain has elapsed since Mixxx
+                    // processed the output signal.
+                    // Although Mixxx receives the input signal delayed, the user hears it mixed
+                    // in hardware with the master & booth outputs without that
+                    // latency, so to record/broadcast the same signal that is heard
+                    // on the master & booth outputs, the master mix must be delayed before
+                    // mixing the talkover signal for the record/broadcast mix.
+                    // If not using microphone inputs or recording/broadcasting from
+                    // a sound card input, skip unnecessary processing here.
+
+                    // Copy the master mix to a separate buffer before delaying it
+                    // to avoid delaying the master output.
+                    m_pLatencyCompensationDelay->process(m_pSidechainMix, m_iBufferSize);
+                    SampleUtil::add(m_pSidechainMix, m_pTalkover, m_iBufferSize);
+                }
             }
         }
 
-        // Submit buffer to the side chain to do broadcasting, recording,
-        // etc. (CPU intensive non-realtime tasks)
-        // If recording/broadcasting from a sound card input,
-        // SoundManager will send the input buffer from the sound card to m_pSidechain
-        // so skip sending a buffer to m_pSidechain here.
-        if (!m_bExternalRecordBroadcastInputConnected
-            && m_pEngineSideChain != nullptr) {
+        // Submit buffer to the side chain to do CPU intensive non-realtime
+        // tasks like recording. The SoundDeviceNetwork, responsible for
+        // passing samples to the network reads directly from m_pSidechainMix,
+        // registering it with SoundDevice::addOutput().
+        // Note: In case the broadcast/recording input is configured,
```

Comment:
What does this note mean? Why are you referencing a condition that you don't check in the code? m_pEngineSideChain->writeSamples() is invoked unconditionally.

Reply:
I told him to make that more explicit. Currently writeSamples doesn't care about the recordinput, but that's because now the buffer (psidechainmix) is filled elsewhere and that's what the comment tries to explain

Reply:
Do you have an idea to Improve the situation? 

I think the root cause of this confusion is that the EnginSideChain combines some unrelated features. Driving the side chain tasks, collecting the samples for recording and receiving the sampled from the sound card input. 

A good solution would be to split this up and turn it into a sound device. 
This is however to much work for this simple bug fix.